### ISSUE TITLE
unclear how to install public plugins in self hosted edition

### ISSUE THREAD

**Issue by glowingkitty** [uxbox/uxbox#5269]
Since the https://penpot.app/penpothub/plugins page only links to the cloud version of penpot and doesn't show the required link to the manifest file of the plugin - how can we install public plugins in the self hosted edition?
Also, from a UX perspective - the plugins store should be available from penpot directly. Coming in the next update?

**Comment by aandaz** [comment]
I am facing the same issue. I did try to follow a few things but still could not get it to work. 

I tried changing the URL to the self-hosted instance 

https://{{MY SELFHOSTED INSTANCE}} /#?plugin=https://qrcode-penpot-plugin.ap.cx/manifest.json

I get the installation window but when I open my Penpot Project there is no plugin icon or anyway to access the plugin.

The plugin manager interface is nowhere to be found.

I even tried the shortcuts that were mentioned in the plugin docs. No luck with that too.

**Comment by mbtechtt** [comment]
I am also unable to use plugins on the self-hosted version of penpot. I have been trying to see if I could figure out what is causing the issue. But so far I have seen that there is the shortcut for plugins in the shortcuts menu (pictured below). Also pictured below is the missing Plugin option of the dropdown menu.

![image](https://github.com/user-attachments/assets/f735d1d9-6e85-44a9-bca5-96fbd5023f8c)

> I tried changing the URL to the self-hosted instance
> 
> https://{{MY SELFHOSTED INSTANCE}} /#?plugin=https://qrcode-penpot-plugin.ap.cx/manifest.json
> 
> I get the installation window but when I open my Penpot Project there is no plugin icon or anyway to access the plugin.

I also would have tried what @aandaz tried in changing the URL and got the installation window, but I am also unable to access the plugins menu or options even with the keyboard shortcut. I have tried both Chromium and Firefox Browsers and have not been able to access the menu. Using the DevTools in Firefox, I was able to see that there is indeed a `plugins-runtime/index.js` file being loaded in the HTML head. And that a warning is being produced about the absence of the absence of the source map file. I have previously seen this error on webpages and have them work fine, but I am still mentioning it in case it can help. (image included below)

> Source map error: Error: request failed with status 404
> Resource URL: https://{{MY_INSTANCE_URL}}/plugins-runtime/index.js?ts=1731326189565
> Source Map URL: index.js.map

![image](https://github.com/user-attachments/assets/17a80883-98aa-4a00-bf01-d7d5f7c70f87)

**Comment by mbtechtt** [comment]
I did a screen capture of the plugin "install process".

[Screencast_20241114_132531.webm](https://github.com/user-attachments/assets/d64aa270-e01a-42a6-bc0c-bb8219ee6720)

After it gets to that screen nothing shows up and in the console an error shows up:

![image](https://github.com/user-attachments/assets/3da71553-7f72-4d47-a920-f7200768ef67)

**Comment by mbtechtt** [comment]
@aandaz Are you able to access the plugin's manager now?
I just got back around to trying out Penpot as my internet was down over the weekend and now that I am able to access my instance and I am able to see the plugins manager and menu option. And all the plugins I added prior by changing the URL to my instance variable appeared in the list.

If not, it is worth mentioning that I would have gone through the release notes menu (so you can do this to see if it resolves your issue). And you should update to the latest version, as a few updates were released in the time between when we originally reported the bug and now.

* * *

As for the issue of an easier way to install plugins to self-hosted instances. I have a few ideas that could be used to make this process simpler:

1. When clicking the "Discover more plugins"
![image](https://github.com/user-attachments/assets/86063506-09e8-4a19-bdc5-92862775238a)
It could load the URL with a query string, or other method to pass the instance URL. (It is possible some persons may worry about being tracked with this option) i.e "https://penpot.app/penpothub/plugins?instance={{ INSTANCE_URL }}"

2. Add an input text field that could possibly be saved to browser localstorage the instance URL that is used for the "Install" buttons on the https://penpot.app/penpothub/plugins website.

**Comment by aandaz** [comment]
Updated the self hosted instance now I can see the plugin manager.

**Comment by TheRolfFR** [comment]
I like @BossMBGaming idea, you could do just like Home Assistant does with a button:
![image](https://github.com/user-attachments/assets/d4900836-6d73-40c5-9886-68021f26e8f9)
And have a page to redirect to yout penpot instance and store that URL in LocalStorage:
![image](https://github.com/user-attachments/assets/f6b988c5-9f09-48cf-9de7-0f9a2be0f7d9)

**Comment by chefblogger** [comment]
i installed penpot with docker but i cannot see any installed plugins :( is there a fix to this problem? i use Version 2.3

**Comment by mbtechtt** [comment]
@chefblogger are you using the docker image tagged 2.3.x or the latest tag? Try using the latest tag and see if the plugin menu appears. Also, see if you can access it using the shortcut ctrl/command + alt + p. Plugins are also only accessible through an active project as far as I know.

**Comment by yuleisheng** [comment]
can plugin be used in self hosted penpot? I was able to install a public plugin, but wasn't able to open it.

**Comment by chefblogger** [comment]
yes @yuleisheng  - with the newest version "Version 2.5" i can finaly use plugins without problems

**Comment by yuleisheng** [comment]
@chefblogger congrats! 
I'm using (https://hub.docker.com/layers/penpotapp/frontend/latest/images/sha256-ca8162f6c157585788083a50f9a7ea0a6c919e4899f0c9385595476cf38ddafc), and i can only install but not able to use pluging.

are you using this version (https://hub.docker.com/layers/penpotapp/frontend/2.5.4/images/sha256-ca8162f6c157585788083a50f9a7ea0a6c919e4899f0c9385595476cf38ddafc) ?

**Comment by chefblogger** [comment]
@yuleisheng i use docker as well and at the weekend i updated the latest version with ‘docker compose -f docker-compose.yaml pull’ - i used this tutorial: https://help.penpot.app/technical-guide/getting-started/docker/

which plugin are you having problems with? tell me the name and I'll test everything on my end

**Comment by xychong** [comment]
@chefblogger were you able to install and use plugins on docker without internet?

**Comment by mbtechtt** [comment]
> I like @BossMBGaming idea, you could do just like Home Assistant does with a button: ![image](https://github.com/user-attachments/assets/d4900836-6d73-40c5-9886-68021f26e8f9) And have a page to redirect to yout penpot instance and store that URL in LocalStorage: ![image](https://github.com/user-attachments/assets/f6b988c5-9f09-48cf-9de7-0f9a2be0f7d9)

I would like to reiterate the importance of implementing something like this for persons who self-host penpot. I am unsure if there is a different repo that this request should be sent to as it isn't exactly what the original issue was asking about. But I second this idea.

**Comment by TheRolfFR** [comment]
> I would like to reiterate the importance of implementing something like this for persons who self-host penpot. I am unsure if there is a different repo that this request should be sent to as it isn't exactly what the original issue was asking about. But I second this idea.

You can always make a new issue and reference this one
